presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt

```diff
@@ -87,110 +81,219 @@ fun EventScreen(
+                                }
+                            }
+                        )
+                    }
 
-                    AlarmReminder(
-                        reminderText = stringResource(id = eventScreenState.alarmReminderItem.stringResId),
-                        modifier = Modifier
-                            .fillMaxWidth()
-                            .background(color = MaterialTheme.colorScheme.backgroundWhiteColor),
-                        isEditMode = eventScreenState.isEditMode,
-                        onReminderClicked = {
-                            eventScreenEvent(EventScreenEvent.OnShowAlarmReminderDropdown(shouldOpen = true))
-                            bottomSheetScope.launch {
-                                bottomSheetState.show()
+                    item {
+                        Spacer(modifier = modifier.height(26.dp))
+                        VisitorFilter(
+                            modifier = Modifier.fillMaxWidth(),
+                            selectedVisitorType = eventScreenState.selectedVisitorFilterType,
+                            onSelectedTypeClicked = { visitorFilterType ->
+                                eventScreenEvent(
+                                    EventScreenEvent.OnVisitorFilterTypeChanged(
+                                        visitorFilterType
+                                    )
+                                )
+                            },
+                            onAddVisitorClicked = {
+                                eventScreenEvent(
+                                    EventScreenEvent.OnShowVisitorDialog(
+                                        shouldShowVisitorDialog = true
+                                    )
+                                )
                             }
+                        )
+                    }
+
+                    if(eventScreenState.selectedVisitorFilterType == VisitorFilterType.ALL || eventScreenState.selectedVisitorFilterType == VisitorFilterType.GOING) {
+                        /** Going section */
+                        item {
+                            Text(
+                                text = stringResource(id = R.string.going),
+                                fontWeight = FontWeight.Medium,
+                                fontSize = 16.sp,
+                                color = MaterialTheme.colorScheme.visitorTextFontColor
+                            )
+                            Spacer(modifier = Modifier.height(16.dp))
+
                         }
-                    )
-                    Spacer(modifier = modifier.height(26.dp))
+                        if(eventScreenState.filteredVisitorsGoing.isNotEmpty()) {
+                            items(eventScreenState.filteredVisitorsGoing) { attendee ->
+                                VisitorItem(
+                                    initials = attendee.fullName.toInitials(),
```

Comment:
Here you are calculating .toIntials on every recomposition

This app won't have much performance issues, but in the future, you could have it calculated on the ViewModel just once, and use that list

Reply:
@MKiperszmid Updated